src/components/ArticleGrid.jsx

```diff
@@ -11,12 +11,12 @@ export default function ArticleGrid({ topic }) {
   const [articles, setArticles] = useState([]);
   useEffect(() => {
     setIsLoading(true);
-    getArticles(page, pageLimit).then(({ articles, total_count }) => {
+    getArticles(page, pageLimit, topic).then(({ articles, total_count }) => {
```

Comment:
Excellent - great to see you're using the original api call and using a topic value to fetch the data by filtering with a query